src/Compilers/Core/Portable/Serialization/StreamObjectReader.cs

```diff
@@ -0,0 +1,1020 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+
+namespace Roslyn.Utilities
+{
+    using EncodingKind = StreamObjectWriter.EncodingKind;
+    using Variant = StreamObjectWriter.Variant;
```

Comment:
It was easier to share across libraries by having less implementation details strung out into separate linked files.  Secondly, since all consumers are currently internals visible to (since they are internal APIs) I wanted to have some modicum of implementation detail hiding.  Nesting the implementation types was the best I can do.